core/rawdb/freezer.go

```diff
@@ -308,6 +315,31 @@ func (f *freezer) Sync() error {
 	return nil
 }
 
+// validate checks that every table has the same length.
+// Used instead of `repair` in readonly mode.
+func (f *freezer) validate() error {
+	if len(f.tables) == 0 {
+		return nil
+	}
+	var length uint64
+	var name string
+	// Hack to get length of any table
+	for k, table := range f.tables {
```

Comment:
I think we can merge these two loops into single one, to have less code.

Reply:
Other nit, please rename `k` to something a bit better, lest go with `kind` without a better idea.

Reply:
> I think we can merge these two loops into single one, to have less code.

I understand that would be shorted in terms of lines, but I find it more readable to have them logically separate. If you don't have a strong feeling about it I'd like to keep it as is.